fix: add .def files to textual_hdrs of clang targets

adincebic opened the pull request:
Some packages like [KSCrash](https://github.com/kstenerud/KSCrash) rely on .def files as textual headers.
I first noticed this while investigating https://github.com/cgrindel/rules_swift_package_manager/issues/1279

cgrindel reviewed (commented):
I would prefer that we have a test and/or example that exercises this functionality. In an example, could we patch the KSCrash `Package.swift` to [link the zlib library](https://github.com/cgrindel/rules_swift_package_manager/issues/1279#issuecomment-2855401573)?

adincebic commented:
> I would prefer that we have a test and/or example that exercises this functionality. In an example, could we patch the KSCrash `Package.swift` to [link the zlib library](https://github.com/cgrindel/rules_swift_package_manager/issues/1279#issuecomment-2855401573)?

Makes sense, gonna do that later today.

adincebic commented:
@cgrindel what should I do to make CI green? Looks like I am missing something obvious

cgrindel commented:
> @cgrindel what should I do to make CI green? Looks like I am missing something obvious

It looks like there was a `.build/...` directory in your source tree when you ran `//:tidy`. It added the contents of that directory which included a sub-directory that had a space in the name. I cleared the deleted packages values in `.bazelrc` and ran `//:tidy`, again.

cgrindel reviewed (commented):
Overall, looks good.  A few comments and we should be be good to go.

adincebic commented on examples/kscrash_example/Sources/MyApp/MyApp.swift:
Those are spaces already. But I'll run swiftformat over this file.

adincebic commented:
@cgrindel I was able to get them to merge my change that explicitly links against zlib so I removed the patch and updated the dependency. See https://github.com/kstenerud/KSCrash/pull/627
